core/src/main/java/bisq/core/dao/governance/proposal/param/ChangeParamValidator.java

```diff
@@ -214,24 +222,64 @@ private void validateParamValue(Param param, String inputValue, int chainHeight)
             case PHASE_VOTE_REVEAL:
             case PHASE_BREAK3:
                 if (isMainnet)
-                    checkArgument(newValueAsBlock >= 6, "The break must have at least 6 blocks");
+                    checkArgument(inputValueAsBlock >= 6, "The break must have at least 6 blocks");
                 break;
             case PHASE_RESULT:
                 if (isMainnet)
-                    checkArgument(newValueAsBlock >= 1, "The break must have at least 1 block");
+                    checkArgument(inputValueAsBlock >= 1, "The break must have at least 1 block");
                 break;
+        }
+    }
+
+    private void validateAddressValue(String currentParamValue, String inputValue) {
+        checkArgument(!inputValue.equals(currentParamValue), "Your input must be different to the current value");
+        InputValidator.ValidationResult validationResult = new BtcAddressValidator().validate(inputValue);
+        if (!validationResult.isValid)
+            throw new AddressFormatException(validationResult.errorMessage);
+    }
 
-            default:
-                log.warn("Param {} not handled in switch case at validateParamValue", param);
+    private static void validationChange(double currentParamValue, double inputValue, Param param) throws ParamValidationException {
+        Result result = getChangeValidationResult(currentParamValue,
+                inputValue,
+                param.getMaxDecrease(),
+                param.getMaxIncrease());
+        if (result != Result.OK) {
+            throw new ParamValidationException(result);
         }
     }
 
-    private void validateChangeRange(double currentValue, double newValue, double min, double max) throws ValidationException {
+    /**
+     *
+     * @param currentValue      Current value
+     * @param newValue          New value
+     * @param min               Decrease of param value limited to current value / maxDecrease. If 0 we don't apply the check and any change is possible
+     * @param max               Increase of param value limited to current value * maxIncrease. If 0 we don't apply the check and any change is possible
+     * @return Validation result
+     */
+    @VisibleForTesting
+    static Result getChangeValidationResult(double currentValue, double newValue, double min, double max) {
+        checkArgument(min >= 0, "Min must be >= 0");
+        checkArgument(max >= 0, "Max must be >= 0");
+        if (currentValue == newValue)
+            return Result.SAME;
+
+        if (max == 0)
+            return Result.OK;
```

Comment:
Are you assuming that if max is zero, then so is min? If max is zero but min is non-zero (not sure if there is a use case for that) and the change is <1, then it wont validate min.

Reply:
Yes that is not correct. Was a bit too lazy... Basically there is no use case for setting one to 0 and the other not. But should handle it correclty... The whole validation code could need more love...